text/0017-lmdb-vs-leveldb.md

```diff
@@ -0,0 +1,238 @@
+
+However, a review of the <a href="https://groups.google.com/forum/#!topic/leveldb/GXhx8YvFiig">LevelDB Issue 397 Workaround</a> discussion and some of the <a href="https://github.com/google/leveldb/issues?utf8=%E2%9C%93&q=corrupt+is%3Aissue">issues containing the string "corrupt"</a> in LevelDB's issue tracker suggests that corruption issues are taken seriously and actively investigated.
+
+### Storage Footprint
+
+In a run of the disk space "benchmark" in the [mykmelez/kvbench](https://github.com/mykmelez/kv-bench/) repo, LMDB used roughly 3.5–4x more disk space than LevelDB to store equivalent amounts of data. However, that benchmark employs a hack to measure the space taken by the LMDB and LevelDB datastores on disk, and it isn't clear that the hack is an accurate way to measure disk usage.
+
+#### LMDB
+
+LMDB doesn't compress data on disk, and it doesn't compact datastores, but it does reuse the space freed by deleted entries.
+
+#### LevelDB
+
+LevelDB compresses data on disk using <a href="https://github.com/google/snappy">Google's Snappy compression library</a>, and it compacts datastores.
+
+## References
+
+[LMDB: The Leveldb Killer?](http://banksco.de/p/lmdb-the-leveldb-killer.html)
+
+[Is LMDB a LevelDB Killer?](https://symas.com/is-lmdb-a-leveldb-killer/)
+
+[Understanding LMDB Database File Sizes and Memory Utilization](https://docs.google.com/document/d/367qDPvx60GTNF8XqLtFv8FTNYvpcYQ_Wb9kqO-MKnaI/) (The link is to a Google Docs copy of the [Google cache of the article](https://webcache.googleusercontent.com/search?q=cache:MBZhGNmOXTkJ:https://symas.com/understanding-lmdb-database-file-sizes-and-memory-utilization/+&cd=3&hl=en&ct=clnk&gl=us&client=firefox-b-3-ab), as the article content is currently hidden in the [original page](https://symas.com/understanding-lmdb-database-file-sizes-and-memory-utilization/).)
+
+[LevelDB "impl" document](https://github.com/google/leveldb/blob/master/doc/impl.md)
+
+## Notes
+
+### Footnote 1
+
+<sup>1</sup>See Wikipedia's <a href="https://en.wikipedia.org/wiki/Reliability_of_Wikipedia">Reliability of Wikipedia</a> article for much discussion about the reliability of Wikipedia articles generally. This document uses the Wikipedia articles on LMDB and LevelDB as a source—but not the sole source—of information about the two storage engines.
+
+### Footnote 2
+
+<sup>2</sup>700,380 bytes for the LMDB program versus 639,836 bytes for the control program on a macOS system is an 80,544 byte difference. Both programs built with optimizations enabled.
```

Comment:
Stripping symbols will make more difference than optimizations.  Stripped?  Maybe include the Rust `[profile]` block, or link to it?  These comparisons are inherently difficult to reproduce, I know :)

Reply:
I rebuilt and stripped the executables, which changed the numbers a bit but didn't alter the relative differences much. I don't have a [profile] block, but I added to the footnotes the Rust version and a note that I used the "release" profile and stripped the executables.